File: python/mod_8/task_6.py
def main():
    print("=====Маятник=====")
    while True:
        full_amplitude = float(input("Введите начальную (полную) амплитутду маятника в см: "))
        final_amplitude = float(input("Введите конечную амплитутду маятника, при которой будет считаться, что он остановился, в см: "))
        i = 0

        while full_amplitude > final_amplitude:
            full_amplitude -= full_amplitude * 0.084
            i +=1

        print(f"Перед полным затуханием маятник качнется {i} раз")

if __name__ == "__main__":
    main()

File: final/mod_8/task_6.cpp
#include <iostream>

int main(){
    using namespace std;
    cout << "=====Маятник=====\n";
    while (true){
        float fullAmplitude, finalAmplitude;
        int i = 0;
        do {
            cout << "Введите начальную (полную) амплитутду маятника в см: "; cin >> fullAmplitude;
            if (fullAmplitude < 0) cout << "Начальная амплитуда не может быть меньше нуля. \n";
        } while (fullAmplitude < 0);

        do {
            cout << "Введите конечную амплитутду маятника, при которой будет считаться, что он остановился, в см: "; cin >> finalAmplitude;
            if (finalAmplitude > fullAmplitude || finalAmplitude < 0) cout << "Конечная амплитуда не может быть больше начальной или меньше нуля.\n";
        } while (finalAmplitude > fullAmplitude || finalAmplitude < 0);

        while (fullAmplitude > finalAmplitude){
            fullAmplitude -= fullAmplitude * 0.084;
            i++;
        }
        cout << "Перед полным затуханием маятник качнется "<< i << " раз\n";
        cout << "==========\n";
    }

}